pkg/selector/outputs/bubbletea.go

```diff
@@ -0,0 +1,282 @@
+	colGPU                = "GPUs"
+	colGPUMemory          = "GPU Memory (GiB)"
+	colGPUInfo            = "GPU Info"
+	colODPrice            = "On-Demand Price/Hr"
+	colSpotPrice          = "Spot Price/Hr (30 day avg)"
+
+	// controls
+	controlsString = "Controls: ↑/↓ - up/down • ←/→  - left/right • shift + ←/→ - pg up/down • q - quit"
+)
+
+var (
+	customBorder = table.Border{
+		Top:    "─",
+		Left:   "│",
+		Right:  "│",
+		Bottom: "─",
+
+		TopRight:    "╮",
+		TopLeft:     "╭",
+		BottomRight: "╯",
+		BottomLeft:  "╰",
+
+		TopJunction:    "┬",
+		LeftJunction:   "├",
+		RightJunction:  "┤",
+		BottomJunction: "┴",
+		InnerJunction:  "┼",
+
+		InnerDivider: "│",
+	}
+)
+
+// BubbleTeaModel is used to hold the state of the bubble tea TUI
+type BubbleTeaModel struct {
+	// the model for the table output
+	TableModel table.Model
+}
+
+// NewBubbleTeaModel initializes a new bubble tea Model which represents
+// a stylized table to display instance types
+func NewBubbleTeaModel(instanceTypes []*instancetypes.Details) BubbleTeaModel {
+	return BubbleTeaModel{
+		TableModel: createTable(instanceTypes),
+	}
+}
+
+// Init is used by bubble tea to initialize a bubble tea table
+func (m BubbleTeaModel) Init() tea.Cmd {
+	return nil
+}
+
+// Update is used by bubble tea to update the state of the bubble
+// tea model based on user input
+func (m BubbleTeaModel) Update(msg tea.Msg) (tea.Model, tea.Cmd) {
+	// check for quit
+	switch msg := msg.(type) {
+	case tea.KeyMsg:
+		switch msg.String() {
+		case "ctrl+c", "esc", "q":
+			return m, tea.Quit
+		}
+	case tea.WindowSizeMsg:
+		// handle screen resizing
+
+		// This is needed to handle a bug with bubble tea
+		// where resizing causes misprints (https://github.com/Evertras/bubble-table/issues/121)
+		termenv.ClearScreen()
+
+		// handle width changes
+		m.TableModel = m.TableModel.WithMaxTotalWidth(msg.Width)
+
+		// handle height changes
+		if headerAndFooterPadding >= msg.Height {
+			// height too short to fit rows
+			m.TableModel = m.TableModel.WithPageSize(0)
+		} else {
+			newRowsPerPage := msg.Height - headerAndFooterPadding
+			m.TableModel = m.TableModel.WithPageSize(newRowsPerPage)
+		}
+	}
+
+	// update table
+	var cmd tea.Cmd
+	m.TableModel, cmd = m.TableModel.Update(msg)
+
+	// update footer
+	controlsStr := lipgloss.NewStyle().Faint(true).Render(controlsString)
+	footerStr := fmt.Sprintf("Page: %d/%d | %s", m.TableModel.CurrentPage(), m.TableModel.MaxPages(), controlsStr)
+	m.TableModel = m.TableModel.WithStaticFooter(footerStr)
+
+	return m, cmd
+}
+
+// View is used by bubble tea to render the bubble tea model
+func (m BubbleTeaModel) View() string {
+	outputStr := strings.Builder{}
+
+	outputStr.WriteString(m.TableModel.View())
+	outputStr.WriteString("\n")
+
+	return outputStr.String()
+}
+
+// table creation helpers:
+
+// createRows creates a row for each instance type in the passed in list
+func createRows(instanceTypes []*instancetypes.Details) *[]table.Row {
```

Comment:
it would be nice to de-dupe the logic in this function since it's almost identical to the table-wide output.  If we want to update the table-wide output, the interactive view should update automatically.